File: Java/BookList/Book.java
/**
 * @author brian
 *
 * Book class - contains the constructor for the Book object,
 * the author and title
 */
public class Book
{
    private String title; // name of book
    private String author; // name of writer
    
    public Book(String theTitle, String theAuthor)
    {
        title = theTitle;
        author = theAuthor;
    }
    
    // Retrieves the title of the book
    public String getTitle()
    {
        return title;
    }
    
    // Retrieves the author of the book
    public String getAuthor()
    {
        return author;
    }
    
    // Prints the book and author
    public String toString()
    {
        return title + " by " + author;
    }
    
    // Compares two books
    public boolean equals(Book other)
    {
        return title.equals(other.title) && author.equals(other.author);
    }
}

File: C#/BookFilterer/Book.cs
using System;
using BookFilterer;

namespace BookFilterer
{
    /*
        The Book class contains all informations and methods that the Book
        object will use.
    */
    public class Book
    {
        // Instance variables for the Book class
        private String book;
        private String author;

        // Constructor for the Book class.
        // Takes @param String author and
        // String book.
        public Book(String book, String author)
        {
            this.book = book;
            this.author = author;
        }
        
        // GetBook returns the
        // book instance variable
        public String GetBook()
        {
            return book;
        }

        // GetAuthor returns the
        // author instance variable
        public String GetAuthor()
        {
            return author;
        }

        // Overrides the built-in ToString method
        // and prints out the author and book
        public override String ToString()
        {
            return book + " by " + author;
        }

        // IsEquals compares two book objects
        public Boolean IsEquals(Book other)
        {
            return book.Equals(other.book) && author.Equals(other.author);
        }
    }
}